analytics.py

```diff
@@ -0,0 +1,129 @@
+import math
+from assignment_05 import utils
+
+def check_significant(lower, upper, observed_avg):
+    significance = False
+    if(upper - lower <= observed_avg):
+        significance = True
+    return significance 
+
+def compute_critical(listOfAvgNNDistances):
+    criticalPoints = []
+    criticalPoints.append(min(listOfAvgNNDistances))
+    criticalPoints.append(max(listOfAvgNNDistances))
+    return criticalPoints;
+
+def permutations(p):
+    listOfAvgNNDistances = []
+    
+    for x in range(p):
+        tmpList = []
+        tmpList = (utils.create_random_points(100))
+        listOfAvgNNDistances.append(average_nearest_neighbor_distance(tmpList))
+
+    return listOfAvgNNDistances    
+
+def average_nearest_neighbor_distance(points):
+    """
+    Given a set of points, compute the average nearest neighbor.
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+    Returns
+    -------
+    mean_d : float
+             Average nearest neighbor distance
+    References
+    ----------
+    Clark and Evan (1954 Distance to Nearest Neighbor as a
+     Measure of Spatial Relationships in Populations. Ecology. 35(4)
+     p. 445-453.
+    """
+    listOfDistances = []
+    
+    for n in points:
+        lowest = 100
+        for x in points:
+            if utils.euclidean_distance(n,x)!= 0 and utils.euclidean_distance(n,x) < lowest:
+                lowest = utils.euclidean_distance(n,x)
+        listOfDistances.append(lowest)
+
+    totalOfDistances = 0
+
+    for z in listOfDistances:
+        totalOfDistances += z
+
+    mean_d = totalOfDistances/len(points)
+
+    return mean_d
+
+
+def minimum_bounding_rectangle(points):
+    """
+    Given a set of points, compute the minimum bounding rectangle.
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+    Returns
+    -------
+     : list
+       Corners of the MBR in the form [xmin, ymin, xmax, ymax]
+    """
+    minX = 100
+    minY = 100
+    maxX = 0
+    maxY = 0
+

```

Comment:
Here, I would advise using `math.inf` and `-math.inf`.  This way you are not tied to a specific domain.